test_unstructured_ingest/check-num-files-expected-output.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Description: Validate that the number of files in the output directory is as expected.
+#
+# Arguments:
+#   - $1: The expected number of files in the output directory.
+#   - $2: Name of the output folder. This is used to determine the output directory and the expected output directory paths.
+#   - $3: The expected size of the output directory in bytes (e.g. "10k").
+
+set +e
+
+EXPECTED_NUM_FILES=$1
+OUTPUT_FOLDER_NAME=$2
+EXPECTED_SIZE=$3
+SCRIPT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &>/dev/null && pwd)
+EXPECTED_OUTPUT_DIR=$SCRIPT_DIR/expected-structured-output/$OUTPUT_FOLDER_NAME
+NUM_FILES=$(find "$EXPECTED_OUTPUT_DIR" -type f -size +"$EXPECTED_SIZE" | wc -l)
+
+if [[ "$NUM_FILES" -ne "$EXPECTED_NUM_FILES" && "$OVERWRITE_FIXTURES" == "false" ]]; then
```

Comment:
adds "$OVERWRITE_FIXTURES" == "false" (we weren't doing this before, but if we're overwriting I don't think we need to exit here, in fact it's probably more problematic if we do)